lib/node_modules/@stdlib/math/base/special/riemann-zeta/src/main.c

```diff
@@ -0,0 +1,813 @@
+*
+* 3.  For \\(s \in (1,4)\\), we use the approximation
+*
+*     ```tex
+*     \zeta(s) = C + \operatorname{R}(s-n) + \frac{1}{s-1}
+*     ```
+*
+*     with rational approximation \\(\operatorname{R}(z-n)\\), constant \\(C\\), and integer \\(n\\).
+*
+* 4.  For \\(s > 4\\), we use the approximation
+*
+*     ```tex
+*     \zeta(s) = 1 + e^{\operatorname{R}(z-n)}
+*     ```
+*
+*     with rational approximation \\(\operatorname{R}(z-n)\\) and integer \\(n\\).
+*
+* 5.  For negative odd integers, we use the closed form
+*
+*     ```tex
+*     \zeta(-n) = \frac{(-1)^n}{n+1} B_{n+1}
+*     ```
+*
+*     where \\(B_{n+1}\\) is a Bernoulli number.
+*
+* 6.  For negative even integers, we use the closed form
+*
+*     ```tex
+*     \zeta(-2n) = 0
+*     ```
+*
+* 7.  For nonnegative even integers, we could use the closed form
+*
+*     ```tex
+*     \zeta(2n) = \frac{(-1)^{n-1}2^{2n-1}\pi^{2n}}{(2n)!} B_{2n}
+*     ```
+*
+*     where \\(B_{2n}\\) is a Bernoulli number. However, to speed computation, we use precomputed values (Wolfram Alpha).
+*
+* 8.  For positive negative integers, we use precomputed values (Wolfram Alpha), as these values are useful for certain infinite series calculations.
+*
+* ## Notes
+*
+* -   \\(\[\approx 1.5\mbox{e-}8, 1)\\)
+*
+*     -   max deviation: \\(2.020\mbox{e-}18\\)
+*     -   expected error: \\(-2.020\mbox{e-}18\\)
+*     -   max error found (double): \\(3.994987\mbox{e-}17\\)
+*
+* -   \\(\[1,2\]\\)
+*
+*     -   max deviation: \\(9.007\mbox{e-}20\\)
+*     -   expected error: \\(9.007\mbox{e-}20\\)
+*
+* -   \\((2,4\]\\)
+*
+*     -   max deviation: \\(5.946\mbox{e-}22\\)
+*     -   expected error: \\(-5.946\mbox{e-}22\\)
+*
+* -   \\((4,7\]\\)
+*
+*     -   max deviation: \\(2.955\mbox{e-}17\\)
+*     -   expected error: \\(2.955\mbox{e-}17\\)
+*     -   max error found (double): \\(2.009135\mbox{e-}16\\)
+*
+* -   \\((7,15)\\)
+*
+*     -   max deviation: \\(7.117\mbox{e-}16\\)
+*     -   expected error: \\(7.117\mbox{e-}16\\)
+*     -   max error found (double): \\(9.387771\mbox{e-}16\\)
+*
+* -   \\(\[15,36)\\)
+*
+*     -   max error (in interpolated form): \\(1.668\mbox{e-}17\\)
+*     -   max error found (long double): \\(1.669714\mbox{e-}17\\)
+*
+* @param s    input value
+* @return     function value
+*
+* @example
+* double out = stdlib_base_zeta( 1.1 );
+* // returns ~10.584
+*/
+double stdlib_base_zeta( const double s ) {
+	double tmp;
+	double scc;
+	int32_t is;
+	int32_t as;
+	int32_t n;
+	double sc;
+	double r;
+
+	// Check for `NaN`:
+	if ( stdlib_base_is_nan( s ) ) {
+		return 0.0 / 0.0; // NaN
+	}
+
+	// Check for a pole:
+	if ( s == 1.0 ) {
+		return 0.0 / 0.0; // NaN
+	}
+
+	// Check for large value:
+	if ( s >= 56.0 ) {
+		return 1.0;
+	}
+
+	// Check for a closed form (integers):
+	if ( stdlib_base_is_integer( s ) ) {
+		// Cast `s` to a 32-bit signed integer:
+		is = (int32_t)s;
+
+		// Check that `s` does not exceed MAX_INT32:
+		if ( is == s ) {
```

Comment:
Implicitly, we're casting `is` to double here? Is that intentional? If so, should be `(double)is == s`.

Reply:
Yes, I'll update this.